horovod/_keras/__init__.py

```diff
@@ -176,12 +186,14 @@ def allreduce_grads():
                 else:
                     update_ops = []
                 with tf.control_dependencies(update_ops):
-                    return tf.cond(
-                        tf.logical_or(tf.equal(self._aggregation_frequency, 1), tf.equal(
-                            self.counter, self._aggregation_frequency)),
-                        allreduce_grads,
-                        lambda: self.grads,
-                    )
+                    if self._aggregation_frequency > 1:
+                        return tf.cond(
+                            tf.equal(self.counter, self._aggregation_frequency),
+                            allreduce_grads,
+                            lambda: self.grads,
+                        )
+                    else:
+                        return allreduce_grads()
```

Comment:
what is the motivation behind this change?

Reply:
For some reason when running wtth `CapsuleNet` it did not consider `allreduce_grads` and `lambda: self.grads` to be equivalent graphs, and would error out.